### PR TITLE
Add pod API to varlink

### PR DESCRIPTION
**Pull request by haircommander**
Including all of the methods whose functionality is defined by podman pod * .
Also moved the location of getPodStatus to libpod/pod_api to allow it to be used in ListPods and GetPod
Also slightly changed the API specification.

### CODE REVIEW

**Comment by haircommander** [comment]
@jwhonce I changed the pod API a bit, mostly around error handling. All changes are included in just one commit (67 ...). PTAL (also @baude)

**Comment by haircommander** [comment]
@mheon I also moved status to pod_api. I think it's the best way to avoid duplicate code, but also PTAL and let me know what you think.

File: libpod/pod_api.go [hunk 1]
```diff
@@ -392,6 +401,54 @@ func (p *Pod) Status() (map[string]ContainerStatus, error) {
 	return status, nil
 }
 
+// Status determines the status of the pod based on the
```
Comment (mheon) [inline libpod/pod_api.go]: I don't think this belongs here... We really need a package for code shared between Varlink and cmd/podman. Maybe rename cmd/podman/batchcontainer to cmd/podman/shared and stick it in there?

Comment (haircommander) [reply libpod/pod_api.go]: I can put getPodStatus into batchcontainer for now, then spin up a second pr for the name switch? WDYT @mheon

Comment (mheon) [reply libpod/pod_api.go]: I think it being part of this PR is fine

Comment (haircommander) [reply libpod/pod_api.go]: fixed

Comment (haircommander) [reply libpod/pod_api.go]: @mheon now that I look at it, should it be moved to pkg/ instead of cmd/podman ?

Comment (mheon) [reply libpod/pod_api.go]: I think cmd/podman/ is still a good place for now.

File: cmd/podman/varlink/io.podman.varlink [hunk 2]
```diff
@@ -756,15 +756,9 @@ error ContainerNotFound (name: string)
...
-# PodNoContainers means the pod does not have any containers associated and cannot be started or restarted.
-error PodNoContainers (name: string)
-
-# PodHasContainers means the pod has containers associated and cannot be stopped without the force boolean set as true.
-error PodHasContainers (name: string)
-
 # PodContainerError means a container associated with a pod failed to preform an operation. It contains
 # a container ID of the container that failed.
-error PodContainerError (names: string)
+error PodContainerError (podname: string, ctrnames: []string)
```
Comment (jwhonce) [inline cmd/podman/varlink/io.podman.varlink]: Other Errors in the VarLink API carry the "reason" for the error as well as details. The "reason" is used in logging/displaying the issue.

Comment (haircommander) [reply cmd/podman/varlink/io.podman.varlink]: fixed:  changed ctrnames to errors, which is a slice of ctr id and reason.

**Comment by baude** [comment]
LGTM

**Comment by mheon** [comment]
Did an extra binary file sneak in? I see a .go.pod.swp

**Comment by haircommander** [comment]
HECK, you're right

File: pkg/varlinkapi/pods.go [hunk 3]
```diff
@@ -0,0 +1,237 @@
...
+	"encoding/json"
+	"syscall"
+
+	"github.com/projectatomic/libpod/cmd/podman/shared"
+	"github.com/projectatomic/libpod/cmd/podman/varlink"
+	"github.com/projectatomic/libpod/libpod"
+)
+
+// CreatePod ...
+func (i *LibpodAPI) CreatePod(call iopodman.VarlinkCall, name, cgroupParent string, labels map[string]string) error {
```
Comment (mheon) [inline pkg/varlinkapi/pods.go]: This should probably accept a struct like container create does - I can see us adding a lot more arguments in the future

File: pkg/varlinkapi/pods.go [hunk 4]
```diff
@@ -0,0 +1,237 @@
...
+func (i *LibpodAPI) UnpausePod(call iopodman.VarlinkCall, name string) error {
+	pod, err := i.Runtime.LookupPod(name)
+	if err != nil {
+		return call.ReplyPodNotFound(name)
+	}
+	ctrErrs, err := pod.Unpause()
+	if err != nil && ctrErrs == nil {
+		return call.ReplyErrorOccurred(err.Error())
+	}
+	if ctrErrs != nil {
```
Comment (mheon) [inline pkg/varlinkapi/pods.go]: This probably deserves its own helper, it gets repeated a fair bit

File: pkg/varlinkapi/pods.go (hunk 3 above)
Comment (haircommander) [reply pkg/varlinkapi/pods.go]: fixed

File: pkg/varlinkapi/pods.go (hunk 4 above)
Comment (haircommander) [reply pkg/varlinkapi/pods.go]: fixed

**Comment by rhatdan** [comment]
@mheon PTAL

**Comment by mheon** [comment]
LGTM
@rh-atomic-bot r+

**Comment by mheon** [comment]
And that would be the merge conflicts from the containers/libpod move

**Comment by mheon** [comment]
@rh-atomic-bot r+

**Pull request closed without merging**